### PR TITLE
Facility name autocomplete

### PR DESCRIPTION
**Pull request by kreek**
Endpoint to allow facilities to be autocompleted by name (filtered by type). Migrations will need to run around merge.

- installs pg_trgm, btree_gin and adds trigram index to base_facilities name column
- adds rake task to import dod facilities to base_facilities from csv
- omits dod facilities from facility locator queries
- adds `GET /v0/facilities/suggested` (maybe `/autocomplete` is better?)

cc @cehsu @cvalarida @markgreenburg

### CODE REVIEW

**Comment by patrickvinograd** [comment]
Is there a need to return the entire facility record for the autocomplete mechanism? Maybe a different serializer that just returns the name or name+any specific attributes needed?

File: app/models/base_facility.rb [hunk 1]
```diff
@@ -116,6 +128,12 @@ def per_page
     def max_per_page
       100
     end
+
+    def suggested_names(facility_types, name_part)
+      BaseFacility.where(
+        facility_type: facility_types.map { |t| TYPE_NAME_MAP[t] }
+      ).where('name ILIKE ?', "%#{name_part}%")
```
Comment (patrickvinograd) [inline app/models/base_facility.rb]: Maybe a conscious business decision, but any consideration for querying on city?

File: app/controllers/v0/facilities/va_controller.rb [hunk 2]
```diff
@@ -23,8 +24,22 @@ def show
...
+           serializer: CollectionSerializer,
+           each_serializer: VAFacilitySerializer
+  end
+
   private
 
+  def validate_types_name_part
+    raise Common::Exceptions::ParameterMissing, 'name_part' if params[:name_part].blank?
+    raise Common::Exceptions::ParameterMissing, 'type' if params[:type].blank?
+    raise Common::Exceptions::InvalidFieldValue.new('type', params[:type]) if
```
[the remark this answers is not in the document]
Comment (kreek) [reply app/controllers/v0/facilities/va_controller.rb]: 👍good catch I'll update it for this scenario

**Comment by kreek** [comment]
@patrickvinograd I had considered a custom serializer, we need name and address for 526. I could limit it to that for now and we can add to it in the future if needed.

**Comment by cvalarida** [comment]
For what it's worth (commenting on a merged PR and all), we don't support filling in other data _yet_, but it's a planned enhancement. Currently, we only need the name, but unless there's a major performance issue, I think it makes sense to send all the information (or at least name and address) even though we won't use it all yet.

**Pull request merged**